### DISCUSSION TITLE
2021-01-14 Runtime & Berry Pickers

### DISCUSSION THREAD

**Discussion by evgenykuzyakov** [near/community#33] [Archived]
# Runtime and Berry Picker Work Group Agenda and Notes

Priorities:
- Standards for fungible token

The scope of the transaction runtime is the following:
* New host methods
* Routing and storage of the receipts and transactions;
    - Work on TX hashes
    - Receipts
* State;
* Fees, refunds, gas price and conversion of tokens to gas, economics
    - TODO: File an DeleteAccount action should send a regular transfer instead of a Refund to create an implicit account.

Applications:
* use-cases from partners
    - TODO: File an issue for a real faucet on Testnet
* needs from application developers
    - Need for a RPC Node and tools to match hardhat/ganache test/dev environment.
    - Follow up on issue for a test RPC node. https://github.com/near/nearcore/issues/3093
* potential improvements for tools
* educational content
    - Bananaswap review - AS contract
    - Pre-contract development - Hack It Friday

Contracts:
* SDK
    - `near-sdk` `2.0.1` - fixes `syn` version issue
    - TODO: CI to run empty project with latest `near-sdk` crate.
* NEAR API JS
* New Contracts
    - Multi-sig PR
* Standards
   - FT is mostly done https://github.com/near/NEPs/issues/141

**Comment by mikedotexe** [comment]
Filling in a bit:
**What was done last week**
- Bananaswap contract review
- Hack-It-Friday pitch
- Reviewed and tracked down `near-sdk-rs` issues with `syn`
- Modification to the EVM runner due to gas issue discovered. Fixed with test coverage
- Working toward having Borsh 0.8.0 upgrade in `near-sdk-rs`, but not completed yet.

A bit light due to internal OKR planning, which will be put on https://gov.near.org as soon as we've assessed the best location for this.

**What is being worked on this coming week**
- Team taking Monday off as observed holiday
- [Fungible token standards are the focus](https://github.com/near/NEPs/issues/141)
- Besides main focus, discovering how to explain Simulation Tests and help on that front
